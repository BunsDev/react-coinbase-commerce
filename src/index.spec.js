import React from 'react';
import CoinbaseCommerceButton from "./index";

test('it renders and matches its snapshot', () => {
  const wrapper  = mount(
    <CoinbaseCommerceButton checkoutId={'aaaa'}/>
  );
  expect(wrapper.children().find('iframe').length).toEqual(0);
  expect(wrapper.children().find('button').length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});

test('it opens up an iframe when its button is clicked', () => {
  const wrapper = shallow(
    <CoinbaseCommerceButton checkoutId={'aaaa'}/>
  );

  wrapper.find('Button').simulate('click');
  expect(wrapper.children().find('IFrame').length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
