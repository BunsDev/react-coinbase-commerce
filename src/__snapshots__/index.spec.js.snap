// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it opens up an iframe when its button is clicked 1`] = `
<React.Fragment>
  <Button
    onClick={[Function]}
  />
  <IFrame
    checkoutId="aaaa"
    onError={[Function]}
    onModalClose={[Function]}
  />
</React.Fragment>
`;

exports[`it renders and matches its snapshot 1`] = `
<CoinbaseCommerceButton
  checkoutId="aaaa"
>
  <Button
    onClick={[Function]}
  >
    <button
      onClick={[Function]}
    >
      Buy With Crypto
    </button>
  </Button>
</CoinbaseCommerceButton>
`;
